raiden-dapp/src/components/ConnectionManager.vue

```diff
@@ -93,22 +110,37 @@ export default class ConnectionManager extends Vue {
    */
   async created(): Promise<void> {
     const { rpc_endpoint: rpcUrl, private_key: privateKey } = await ConfigProvider.configuration();
+
     if (rpcUrl && privateKey) {
       const provider = await DirectRpcProvider.link({ rpcUrl, privateKey });
       this.connect(provider);
     }
   }
 
-  async onProviderLinkEstablished(linkedProvider: EthereumProvider): Promise<void> {
-    await this.connect(linkedProvider);
+  openWalletConnectProviderDialog(): void {
```

Comment:
Question, why not have one function for both displaying and hiding the dialog? `this.walletConnectProviderDialogVisible = !this.walletConnectProviderDialogVisible`? Same for the injected provider? Is it because you want to use the `closeAllProviderDialogs` function?

Reply:
Yes, also. Besides its kinda not correct to say that the action button toggles the visibility and if the cancel event comes it also toggles. This only works when the state is always perfect.
But I agree, I dislike in general that this is so much overhead for each new dialog. I would love to simplify it. I was not able to come up with something that is not ugly in a different manner. For example I was considering to put into the dialog components themselves and there into the base dialog. But its no good practice I think. As long as you don't think this is a genius idea, I would like to stick with the current solution until we have something better.